### PR TITLE
additional information in the ECSOperator around support of launch_type=EXTERNAL

### LINKED ISSUE #21321: ECS Operator does not support launch type "EXTERNAL"
### Description

You can run ECS tasks either on EC2 instances or via AWS Fargate, and these will run in AWS. With ECS Anywhere, you are now able to run the same ECS tasks on any host that has the ECS agent - on prem, in another cloud provider, etc. The control plane resides in ECS, but the execution of the task is managed by the ECS agent. 

To launch tasks on hosts that are managed by ECS Anywhere, you need to specify a launch type of EXTERNAL. This is currently not supported by the ECS Operator. When you attempt do this, you get an error of unsupported launch type.

The current work around is to use boto3 and create a task and then run it using the correct parameters.

### Use case/motivation

The ability to run your task code to support hybrid and multi-cloud orchestration scenarios.

### Related issues

_No response_

### Are you willing to submit a PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by 094459**
I have added some additional information in the ECSOperator around support of launch_type=EXTERNAL, added launch_type EXTERNAL in the test case, and added a new example DAG on how this works.

All tests in Breeze have passed, as have static checks.

closes: https://github.com/apache/airflow/issues/21321

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
How about adding also howto guide with extracted fragment from the example?

**Comment by 094459** [comment]
Yeah, happy to do that. In fact, I have written a whole blog post on this here -> https://dev.to/aws/orchestrating-hybrid-workflows-using-amazon-managed-workflows-for-apache-airflow-mwaa-2boc

Is there anything specific you want me to include, or an existing doc somewhere you want me to change?

**Comment by potiuk** [comment]
We have existing howto guides (and they include parts of the example dags as "exampleinclude"). 

See here: https://github.com/apache/airflow/tree/main/docs/apache-airflow-providers-amazon/operators

**Comment by 094459** [comment]
This looks good - I must have missed this. I will update this doc. From a PR request perspective, do I need to make the doc changes and then make a new PR (and this one will be closed?)

**Comment by potiuk** [comment]
> This looks good - I must have missed this. I will update this doc. From a PR request perspective, do I need to make the doc changes and then make a new PR (and this one will be closed?)

Just add them to the same PR :)

**Comment by 094459** [comment]
I am struggling with the documentation format, as I am typically used to doing all documentation in markdown which is failing in the static code checks. I will go away and read the Sphinx docs and see how I can fix.

File: docs/apache-airflow-providers-amazon/operators/ecs.rst [hunk 1]
```diff
@@ -46,6 +46,149 @@ Before using EcsOperator *cluster*, *task definition*, and *container* need to b
...
+            }
+        ] },
+        tags={
+            "Customer": "X",
+            "Project": "Y",
+            "Application": "Z",
+            "Version": "0.0.1",
+            "Environment": "Development",
+            }
+    )
```
Comment (eladkal) [inline docs/apache-airflow-providers-amazon/operators/ecs.rst]: Code examples are preferably to come from the example dags.
There is no need to hard code them in the doc. Check other operators examples

Comment (094459) [reply docs/apache-airflow-providers-amazon/operators/ecs.rst]: Ok, I did consider this but thought including the whole file was too verbose - I just wanted to include just the specific lines for the sake of clarity. What is the preferred way of doing this? I noticed the partials folder, should I create the code in there instead?

Comment (094459) [reply docs/apache-airflow-providers-amazon/operators/ecs.rst]: Its ok, I am just looking at some of the other Amazon providers (EMR) and it looks like they do something like what I need so I will copy their approach. Thanks.

Comment (094459) [reply docs/apache-airflow-providers-amazon/operators/ecs.rst]: What about IAM Permissions, these are not code and they are specific to the Operator - these feel like they should go in the How To doc. I don't feel they should go in the partials folder as they are not useful in any other doc.

Comment (eladkal) [reply docs/apache-airflow-providers-amazon/operators/ecs.rst]: proceed with how you see fit and we can check it during review.

Comment (094459) [reply docs/apache-airflow-providers-amazon/operators/ecs.rst]: All done, checked in with no errors this time.

**Review by eladkal — APPROVED** [review]

**Pull request merged**